app/util/analytics.py

```diff
@@ -114,24 +125,63 @@ def get_events(since_timestamp=None, since_id=None):
     since_timestamp = float(since_timestamp) if since_timestamp else 0.0
     since_id = int(since_id) if since_id else 0
 
-    with open(_eventlog_file, 'r') as f:
-        # todo(joey): This is inefficient since it reads the whole log file into memory (can be several megabytes).
-        # We probably want something like http://code.activestate.com/recipes/120686/
-        reversed_log_lines = reversed(f.readlines())
+    filtered_events_from_cache = get_events_from_cache(since_timestamp, since_id)
+    if len(filtered_events_from_cache) > 0:
```

Comment:
@josephharrington re: "What if the user requests more than LOG_CACHE_MAX_SIZE?"

It doesn't look like we currently have a way of a user request expressing that they want more than a certain number of entries.  Currently, we support the since_timestamp and since_id query parameters which will return any logs that happened after that date.  Since our logs roll over, there is no guarantee that the response will contain all the logs that occurred after a certain timestamp.  

That is why I do a check if any logs have been returned before falling back to check from the file.